unity-environment/Assets/ML-Agents/Scripts/Academy.cs

```diff
@@ -430,24 +504,31 @@ void EnvironmentStep()
             ConfigureEnvironment();
             modeSwitched = false;
         }
-
         if (isCommunicatorOn)
         {
-            if (communicator.GetCommand() == ExternalCommand.RESET)
+            if (brainBatcher.GetCommand() == 
+                MLAgents.Communicator.Command.Reset)
             {
                 // Update reset parameters.
-                Dictionary<string, float> NewResetParameters =
-                    communicator.GetResetParameters();
-                foreach (KeyValuePair<string, float> kv in NewResetParameters)
+                MLAgents.Communicator.EnvironmentParameters NewResetParameters =
+                            brainBatcher.GetEnvironmentParameters();
+                if (NewResetParameters != null)
                 {
-                    resetParameters[kv.Key] = kv.Value;
+                    foreach (KeyValuePair<string, float> kv in
+                             NewResetParameters.FloatParameters)
+                    {
+                        resetParameters[kv.Key] = kv.Value;
+                    }
                 }
+                SetIsInference(!brainBatcher.GetIsTraining());
 
                 ForcedFullReset();
-                communicator.SetCommand(ExternalCommand.STEP);
             }
-            if (communicator.GetCommand() == ExternalCommand.QUIT)
+            if (brainBatcher.GetCommand() == MLAgents.Communicator.Command.Quit)
             {
+#if UNITY_EDITOR
+                EditorApplication.isPlaying = false;
```

Comment:
Do we really need this? Can we have a helper interface that will remove the "#if" code?

Reply:
We would remove it if we do not want to have the editor training feature. I will try to think of a solution but I think that it is the responsibility of the academy to stop the simulation.